fundingmanager_test.go

```diff
@@ -520,24 +520,62 @@ func openChannel(t *testing.T, alice, bob *testNode, localFundingAmt,
 	return fundingOutPoint
 }
 
-func assertMarkedOpen(t *testing.T, alice, bob *testNode,
-	fundingOutPoint *wire.OutPoint) {
-	state, _, err := alice.fundingMgr.getChannelOpeningState(fundingOutPoint)
-	if err != nil {
-		t.Fatalf("unable to get channel state: %v", err)
+func assertNumPendingChannels(t *testing.T, node *testNode, num int) {
+	var numPendingChans int
+	for i := 0; i < 10; i++ {
+		pendingChannels, err := node.fundingMgr.cfg.Wallet.Cfg.Database.FetchPendingChannels()
+		if err != nil {
+			t.Fatalf("unable to fetch pending channels: %v", err)
+		}
+
+		numPendingChans = len(pendingChannels)
+		if numPendingChans == num {
+			// Success, return.
+			return
+		}
+
+		// Sleep, and try again in a bit.
+		time.Sleep(500 * time.Millisecond)
 	}
 
-	if state != markedOpen {
-		t.Fatalf("expected state to be markedOpen, was %v", state)
+	t.Fatalf("Expected node to have %d pending channels, had %v",
+		num, numPendingChans)
+}
+
+func assertDatabaseState(t *testing.T, node *testNode,
+	fundingOutPoint *wire.OutPoint, expectedState channelOpeningState) {
+
+	var state channelOpeningState
+	var err error
+	for i := 0; i < 10; i++ {
+		state, _, err = node.fundingMgr.getChannelOpeningState(
+			fundingOutPoint)
+		if err != nil && err != ErrChannelNotFound {
+			t.Fatalf("unable to get channel state: %v", err)
+		}
+
+		if state == expectedState {
```

Comment:
`getChannelOpeningState` returns `state` as `markedOpen` if there is an error, so we should not return if `err == ErrChannelNotFound`

Reply:
good catch!